extensions/replication/queueProcessor/QueueProcessor.js

```diff
@@ -287,24 +287,30 @@ class QueueProcessor extends EventEmitter {
     _resumeService(date) {
         const enabled = this._consumer.getServiceStatus();
         const now = new Date();
-        if (date && now < new Date(date)) {
-            // if date is in the future, attempt to schedule job
-            this.scheduleResume(date);
-        } else if (!enabled) {
-            // if currently paused, attempt to resume
-            this._updateZkStateNode('paused', false, err => {
-                if (err) {
-                    this.logger.trace('error occurred saving state to ' +
-                    'zookeeper', {
-                        method: 'QueueProcessor._resumeService',
-                    });
-                } else {
-                    this._consumer.resume(this.site);
-                    this.logger.info('resumed replication for location: ' +
-                        `${this.site}`);
-                    this._deleteScheduledResumeService();
-                }
-            });
+
+        // if currently paused, attempt to resume or schedule resume
+        if (!enabled) {
+            if (date && now < new Date(date)) {
+                // if date is in the future and the site is current paused,
+                // attempt to schedule job
+                this.scheduleResume(date);
+            } else {
+                this._updateZkStateNode('paused', false, err => {
+                    if (err) {
+                        this.logger.trace('error occurred saving state to ' +
+                        'zookeeper', {
+                            method: 'QueueProcessor._resumeService',
+                        });
+                    } else {
+                        this._consumer.resume(this.site);
+                        this.logger.info('resumed replication for location: ' +
+                            `${this.site}`);
+                        this._deleteScheduledResumeService();
+                    }
+                });
+            }
+        } else {
+            this.logger.info(`site ${this.site} is already resumed`);
```

Comment:
It may have been resumed, or it may have never been paused, so may be rephrased to "is not paused" for example.

On a pure style note, I think the error case can be simply checked at the beginning of this function, like an assertion that returns immediately from the function, avoiding the need for a whole indented block for the normal case.